src/snowcli/cli/streamlit/commands.py

```diff
@@ -160,22 +139,31 @@ def streamlit_deploy(
     not exist.
     You can modify the behaviour using flags. For details check help information.
     """
-    if environment_file and not environment_file.exists():
+    dm = DefinitionManager()
+    streamlit = dm.project_definition.get("streamlit")
+    if not streamlit:
+        return MessageResult("No streamlit were specified in project definition.")
+    if streamlit["name"] != streamlit_name:
+        return MessageResult(f"No streamlit {streamlit_name} defined in this project.")
```

Comment:
Assuming there's only one streamlit per project I think we don't need `streamlit_name` flag

Reply:
Good point, removed